src/mmrelay/matrix/sync_bootstrap.py

```
@@ -0,0 +1,1201 @@
+                if isinstance(discovery_response, facade.DiscoveryInfoResponse):
+                    actual_homeserver = discovery_response.homeserver_url
+                    facade.logger.info(
+                        f"Server discovery successful: {actual_homeserver}"
+                    )
+                    homeserver = actual_homeserver
+                elif isinstance(discovery_response, facade.DiscoveryInfoError):
+                    facade.logger.info(
+                        f"Server discovery failed, using original URL: {homeserver}"
+                    )
+                else:
+                    if hasattr(discovery_response, "homeserver_url"):
```

Ruff (0.15.9)

[warning] Use `elif` instead of `else` then `if`, to reduce indentation

Convert to `elif`

(PLR5501)